app/org/sagebionetworks/bridge/stormpath/StormpathAccountDao.java

```diff
@@ -392,9 +392,11 @@ private Account constructAccount(StudyIdentifier studyId, com.stormpath.sdk.acco
         }
         if (healthId == null) {
             healthId = healthCodeService.createMapping(studyId);
+            account.setHealthId(healthId);
             updateAccount(account);
```

Comment:
StormpathAccountDaoMockTest.getAccountCreatesHealthCode() tests this. If you remove that setId now that it's in an if/else statement, the test breaks. If you comment out the other branch, the authenticate test fails because it verifies that we initialize an existing healthCode with a healthId.